codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieDateFormatter.java

```diff
@@ -0,0 +1,452 @@
+ * as it's way more efficient than HttpHeaderDateFormat.
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc6265#section-5.1.1">RFC6265</a> for the parsing side
+ * @see <a href="https://tools.ietf.org/html/rfc1123#page-55">RFC1123</a> for the encoding side.
+ */
+public final class CookieDateFormatter {
+
+    private static final BitSet DELIMITERS = new BitSet();
+    static {
+        DELIMITERS.set(0x09);
+        for (char c = 0x20; c <= 0x2F; c++) {
+            DELIMITERS.set(c);
+        }
+        for (char c = 0x3B; c <= 0x40; c++) {
+            DELIMITERS.set(c);
+        }
+        for (char c = 0x5B; c <= 0x60; c++) {
+            DELIMITERS.set(c);
+        }
+        for (char c = 0x7B; c <= 0x7E; c++) {
+            DELIMITERS.set(c);
+        }
+    }
+
+    private static final ThreadLocal<CookieDateFormatter> INSTANCES = new ThreadLocal<CookieDateFormatter>() {
+        protected CookieDateFormatter initialValue() {
+            return new CookieDateFormatter();
+        }
+    };
+
+    public static Date parse(String txt) {
+        return parse(checkNotNull(txt, "txt"), 0, txt.length());
+    }
+
+    public static Date parse(String txt, int start, int end) {
+        return formatter().parse0(checkNotNull(txt, "txt"), start, end);
+    }
+
+    public static String format(Date date) {
+        return formatter().format0(checkNotNull(date, "date"));
+    }
+
+    public static StringBuilder append(Date date, StringBuilder sb) {
+        return formatter().append0(checkNotNull(date, "date"), checkNotNull(sb, "sb"));
+    }
+
+    private static CookieDateFormatter formatter() {
+        CookieDateFormatter formatter = INSTANCES.get();
+        formatter.reset();
+        return formatter;
+    }
+
+    // delimiter = %x09 / %x20-2F / %x3B-40 / %x5B-60 / %x7B-7E
+    private static boolean isDelim(char c) {
+        return DELIMITERS.get(c);
+    }
+
+    private static boolean isDigit(char c) {
+        return c >= 0x30 && c <= 0x39;
+    }
+
+    private static int getNumericalValue(char c) {
+        return (int) c - 48;
+    }
+
+    private final GregorianCalendar cal = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+    private final StringBuilder sb = new StringBuilder(29); // Sun, 27 Nov 2016 19:37:15 GMT
+    private boolean timeFound;
+    private int hours;
+    private int minutes;
+    private int seconds;
+    private boolean dayOfMonthFound;
+    private int dayOfMonth;
+    private boolean monthFound;
+    private int month;
+    private boolean yearFound;
+    private int year;
+
+    private CookieDateFormatter() {
+        reset();
+    }
+
+    public void reset() {
+        timeFound = false;
+        hours = -1;
+        minutes = -1;
+        seconds = -1;
+        dayOfMonthFound = false;
+        dayOfMonth = -1;
+        monthFound = false;
+        month = -1;
+        yearFound = false;
+        year = -1;
+        cal.set(Calendar.MILLISECOND, 0);
```

Comment:
Agree. I do reset it in `format0` but it would be more consistent to do it in reset?